test/unit/general/deviceService-test.js

```diff
@@ -212,19 +249,69 @@ describe('NGSI-v2 - Device Service: utils', function () {
+        beforeEach(function (done) {
+            contextBrokerMock = nock('http://unexistentHost:1026')
+                .matchHeader('fiware-service', 'testservice')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/NGSI9/registerContext')
+                .reply(
+                    200,
+                    utils.readExampleFile(
+                        './test/unit/examples/contextAvailabilityResponses/registerProvisionedDeviceSuccess.json'
+                    )
+                );
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'testservice')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v1/updateContext')
+                .reply(
+                    200,
+                    utils.readExampleFile('./test/unit/examples/contextResponses/createProvisionedDeviceSuccess.json')
+                );
+
+            async.series([request.bind(request, configGroupCreation), request.bind(request, deviceCreation)], function (
+                error,
+                results
+            ) {
+                done();
+            });
+        });
+
+        it('should return the existing device', function (done) {
+            iotAgentLib.retrieveDevice('Light1', '801230BJKL23Y9090DSFL123HJK09H324HV8732', function (error, device) {
+                should.not.exist(error);
+                should.exist(device);
+
+                device.id.should.equal('Light1');
+                done();
+            });
+        });
+    });
+
+    //FIXME: this test will be removed if at the end /iot/services API (now Deprecated) is removed
     describe('When an unexisting device tries to be retrieved for an existing APIKey', function () {
         beforeEach(function (done) {
             contextBrokerMock = nock('http://unexistentHost:1026')
                 .matchHeader('fiware-service', 'testservice')
                 .matchHeader('fiware-servicepath', '/testingPath')
-                .post('/v2/registrations')
-                .reply(201, null, { Location: '/v2/registrations/6319a7f5254b05844116584d' });
+                .post('/NGSI9/registerContext')
+                .reply(
+                    200,
+                    utils.readExampleFile(
+                        './test/unit/examples/contextAvailabilityResponses/registerProvisionedDeviceSuccess.json'
+                    )
+                );
 
             contextBrokerMock
                 .matchHeader('fiware-service', 'testservice')
                 .matchHeader('fiware-servicepath', '/testingPath')
-                .post('/v2/entities?options=upsert')
-                .reply(204);
+                .post('/v1/updateContext')
+                .reply(
+                    200,
+                    utils.readExampleFile('./test/unit/examples/contextResponses/createProvisionedDeviceSuccess.json')
+                );
```

Comment:
NGSI V1 is no longer supported, Why you included those NGSIv1 tests in NGSIv1 the codebase?